src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -332,6 +339,10 @@ private boolean addNewLogEvent(AtomicInteger totalBytesRead, CloudWatchAttemptLo
                 .isBefore(timestamp)) {
             return true;
         }
+        // Cloudwatch does not allow uploading data older than 14 days.
+        if (timestamp.isBefore(Instant.now(clock).minus(14, ChronoUnit.DAYS))) {
+            return false;
+        }
```

Comment:
should this be the first check in the method? you could return `true` on line 330 even if the timestamp is 14 days before now.